test/test_Time.cpp

```diff
@@ -96,18 +97,54 @@ BOOST_AUTO_TEST_CASE(fromString)
         " vs. converted: " << tzConverted.toString());
     // End time zone check
 
+    base::Time formatNowUTC0 = base::Time::fromString(
+        "2012-06-14--12.05.06Z:001001Z+0 GMT",
+        base::Time::Microseconds,
+        "%Y-%m-%d--%H.%M.%S%Z",
+        true
+    );
+    base::Time formatNowUTC1 = base::Time::fromString(
+        "2012-06-14--12.05.06Z:001001",
+        base::Time::Microseconds,
+        "%Y-%m-%d--%H.%M.%S%Z",
+        true,
+        0
+    );
+
+    BOOST_REQUIRE_EQUAL(formatNowUTC0.toMilliseconds(), formatNowUTC1.toMilliseconds());
```

Comment:
You don't actually test that the conversion is exact. You test that the two calls return the same value - which could be 100% a wrong value. Use a epoch conversion website (I listed one in the comments) and actually figure out what value base::Time#toMicroseconds should have, and then test against that.